lib/strategies/webapp-strategy.js

```diff
@@ -436,9 +437,9 @@ function retrieveTokens(formData, strategy) {
 				let loginError = new Error(loginErrMsg);
 				loginError.statusCode = response && response.statusCode;
 				loginError.code = (body.error || body["error_code"]).toLowerCase();
-				deferred.reject(loginError);
+				return reject(loginError);
 			} catch (e) {
-				deferred.reject(new Error("Failed to obtain tokens"));
+				return reject(new Error("Failed to obtain tokens"));
```

Comment:
you loss the exception here.

Reply:
I've added a log, to be honest, I don't sure why the try and catch is even there, I don't see any way in which this code can throw an exception